docs/intro/index.md

```diff
@@ -4,25 +4,153 @@
+
+## Why OCaml?
+
+OCaml codebases scale well both in terms of quantity of lines and number of
+contributors. The sound type system helps to prevent ambiguous behavior in your
+program---if it compiles, it runs without runtime errors. OCaml and React (via
+[ReasonReact][reasonreact]) are an effective, FP-friendly combination for
+building frontend UIs using JSX syntax. If the backend is also written in OCaml,
+you can share types between the frontend and backend, ensuring that they stay in
+sync[^1].  It's even possible to write [universal React
+components](https://github.com/ml-in-barcelona/server-reason-react) that are
+rendered on the server with the performance of native code[^2].
+
+All that said, how does OCaml compare against other prominent compile-to-JS
+languages?
+
+**TypeScript** is a gradually-typed superset of JavaScript. It's easier to learn
+and adopt because its design highly emphasizes compatibility with JavaScript.
+This brings a few footguns, such as `any`, `as` casting, `ts-nocheck` etc,
+making it less type-safe than OCaml (and all the other languages on this list)[^3].
+
+- Zero-cost JS interop? Yes
+- Can use with React? Yes
+- Supports JSX? Yes
+- Pattern matching? No
+- Can use on server? Yes, via Node.
+
+**ReScript** is very similar to OCaml, even at the syntax level. The main
+difference is that ReScript's only compilation target is JS[^4].
+
+- Zero-cost JS interop? Yes
+- Can use with React? Yes, via [@rescript/react](https://github.com/rescript-lang/rescript-react), which is roughly equivalent to
+  Melange's ReasonReact bindings.
+- Supports JSX? Yes
+- Pattern matching? Yes
+- Can use on server? Yes, via Node.
+
+**Elm** is similar to OCaml. It leans strongly towards pure functional
+programming.
+
+- Zero-cost JS interop? No. Elm treats JavaScript as unsafe, interacting with JS
+  via ports (more like message passing than traditional FFI). Ergonomic
+  integration with React and other libraries isn't possible.
+- Can use with React? No. Elm has its own framework with some similar concepts.
+- Supports JSX? No
+- Pattern matching? Yes
+- Can use on server? No
+
+**F#** is a fullstack language similar to OCaml and is strongly aligned with the
+.NET ecosystem.
+
+- Zero-cost JS interop? Probably
+- Can use with React? Yes. [Feliz](https://github.com/Zaid-Ajaj/Feliz) is an F#
+  wrapper library for React.
+- Supports JSX? No. Feliz uses ordinary function call syntax for render logic.
+- Pattern matching? Yes
+- Can use on server? Yes, via .NET or Node.
+
+**Gleam** is a fullstack language similar to OCaml which is aligned with the
+BEAM ecosystem.
+
+- Zero-cost JS interop? No, but Gleam's JS compilation target is relatively new,
+  so it may become more optimized in the future.
+- Can use with React? Yes, via [redraw](https://github.com/ghivert/redraw) and [react-gleam](https://github.com/brettkolodny/react-gleam).
+- Supports JSX? No
+- Pattern matching? Yes
```

Comment:
I don't see the point of this section, this is a book of Reason and Melange. 

Let's encourage people to read about comparisons in the melange documentation or a blog post.

Reply:
I'd kind of lean this way too. It reads a bit like a sales pitch. Maybe at a minimum, cut down on a few of the alternatives.